docs_nnx/guides/transforms.md

```diff
@@ -302,14 +302,44 @@ def noisy_vector_dot(weights: Weights, x: jax.Array):
   y = x @ weights.kernel + weights.bias
   return y + random.normal(weights.rngs.noise(), y.shape)
 
-y1 = noisy_vector_dot(weights, x)
-y2 = noisy_vector_dot(weights, x)
+y1 = noisy_vector_dot(weights, count)
+y2 = noisy_vector_dot(weights, count)
 
 print(jnp.allclose(y1, y2))
 nnx.display(weights)
 ```
 
-## Consistent aliasing
+## Rules and limitations
+In this section we will cover some rules and limitations apply when using Modules inside transformations.
+
+### Mutable Module cannot be passed by closure
+
+While Python allows for passing objects as closures to functions, this is generally not supported by Flax NNX transforms. The reason is that because Modules are mutable, it is very easy to insert a tracer into a captured Module and get tracer leakage. To avoid this, Flax NNX Module and Variables check that the current context is valid every time they are performing an update operation.
+
+For example, if we a have stateful Module such as `Counter` that increments a counter every time it is called, and we try to pass it as a closure to a function decorated with `nnx.jit`, we would be leaking the tracer, however Flax NNX will raise an error instead to prevent this:
+
```

Comment:
I think it's worth to add an additional line saying this can be solved by explicitly pass the module `counter` as an argument of `f`.

Reply:
Good idea. I simplified the language and added a comment add the end with the solution.